generators/entity-client/templates/angular/src/main/webapp/app/entities/update/entity-management-update.component.spec.ts.ejs

```diff
@@ -124,6 +129,79 @@ _%>
+
+        describe('Loading relationships', () => {
+            describe('loadRelationshipsOptions', () => {
+    <%_ for (const relationshipsByEntityNeedingOptions of allRelationshipsByEntityNeedingOptions) { _%>
+        <%_ const relationshipsWithCustomUniqueOptions = relationshipsByEntityNeedingOptions.filter(rel => rel.relationshipType === 'one-to-one' && rel.otherEntityName !== 'user'); %>
+        <%_ const relationshipsWithCustomSharedOptions = relationshipsByEntityNeedingOptions.filter(rel => !relationshipsWithCustomUniqueOptions.includes(rel)); %>
+        <%_ if (relationshipsWithCustomSharedOptions.length > 0) { _%>
+            <%_ const otherEntity = relationshipsByEntityNeedingOptions[0].otherEntity _%>
+
+                it('Should call <%= otherEntity.entityAngularName %> query and add missing value', () => {
+                    fakeAsync(() => {
+                        const <%= entityInstance %> : I<%= entityAngularName %> = <%- generateTestEntityPrimaryKey(primaryKey, 0) %>;
+            <%_ for (const relationship of relationshipsWithCustomSharedOptions) { _%>
+                <%_ const reference = relationship.reference _%>
+                <%_ if (relationship.collection) { _%>
+                        const <%= reference.name %> : I<%= otherEntity.entityAngularName %>[] = [<%- generateTestEntityPrimaryKey(otherEntity.primaryKey) %>];
+                <%_ } else { _%>
+                        const <%= reference.name %> : I<%= otherEntity.entityAngularName %> = <%- generateTestEntityPrimaryKey(otherEntity.primaryKey) %>;
+                <%_ } _%>
+                        <%= entityInstance %>.<%= reference.name %> = <%= reference.name %>;
+            <%_ } _%>
+                        comp.updateForm(<%= entityInstance %>);
+
+                        const <%= otherEntity.entityInstance %>Colection: I<%= otherEntity.entityAngularName %>[] = [<%- generateTestEntityPrimaryKey(otherEntity.primaryKey) %>];
+                        spyOn(<%= otherEntity.entityInstance %>Service, 'query').and.returnValue(of(new HttpResponse({ body: <%= otherEntity.entityInstance %>Colection })));
+                        const additional<%= otherEntity.entityAngularNamePlural %> = [
+            <%_ for (const relationship of relationshipsWithCustomSharedOptions) { _%>
+                <%_ const reference = relationship.reference _%>
+                            <% if (relationship.collection) { %>...<% } %><%= reference.name %>,
+            <%_ } _%>
+                        ];
+                        const expectedColection: I<%= otherEntity.entityAngularName %>[] = [...additional<%= otherEntity.entityAngularNamePlural %>, ...<%= otherEntity.entityInstance %>Colection];
+                        spyOn(<%= otherEntity.entityInstance %>Service, 'add<%= otherEntity.entityAngularName %>ToCollectionIfMissing').and.returnValues(expectedColection);
+
+                        comp['loadRelationshipsOptions']();
```

Comment:
```suggestion
                        comp.ngOnInit();
```
It's a good practice to test public interface. Can you give input to component through `ActivatedRoute` and run public `ngOnInit()` instead?

Reply:
Not an expert on angular tests.
So can take some time.

Reply:
Sample of providing `ActivatedRoute` is for example in https://github.com/jhipster/generator-jhipster/blob/main/generators/entity-client/templates/angular/src/main/webapp/app/entities/detail/entity-management-detail.component.spec.ts.ejs#L44-L49